### ISSUE TITLE
Monads not working in B28x ?

### ISSUE THREAD

**Issue by DolphinDream** [nortikin/sverchok#3074]
Are monads not fully supported in B28x yet?

**Comment by DolphinDream** [comment]
Here’s a demo.. a simple setup with 3 number nodes grouped into a Monad. Once you connect the 3 nodes to the monad input sockets, exit the monad and change the value of one input all inputs values change to same value.

[![](http://img.youtube.com/vi/AxLFCXdnnaI/0.jpg)](http://www.youtube.com/watch?v=AxLFCXdnnaI "Monad Issue")

**Comment by zeffii** [comment]
yep. i regret that this is still an issue, so some time ago i added a new operator that can make a new socket with min/max etc.

see this thread for a collection of issues.  https://github.com/nortikin/sverchok/issues/2886

n particular the workaround for now is here:
https://github.com/nortikin/sverchok/pull/2893

**Comment by zeffii** [comment]
it's not that monad is broken, it's more that monad was never finished properly.

i'm tempted to fix monad by rewriting it entirely, and we'll just have to see how old `.blends` will be handled.

**Comment by zeffii** [comment]
if I get some time today i'll take another stab at it, from scratch. I'll leave this issue open maybe it will piss me off enough to dig in.

**Comment by Durman** [comment]
> i'm tempted to fix monad by rewriting it entirely,

If so I would request to consider opportunity of creating UI which let choose a node tree which should be evaluated by a monad. 

I think in some places it can make code simpler. Also such approach would be quite Blenderic. There are a lot of examples where Blender let select objects which should be related with another object. For example object and mesh or image texture node and image. And it defenetly would let make work in Sverchok more flexible. Probably separate issue worth creating.

**Comment by zeffii** [comment]
i think indeed i will let user pick `node_tree` from list of `monad_node_trees`, directly from the UI.

**Comment by zeffii** [comment]
two things will need to happen before I can tackle this.

1.  must have multiple hours without distraction
2.  must feel like doing this during that time

1 and 2 do not occur concurrently on a regular basis.

**Comment by zeffii** [comment]
having this open is not so helpful
